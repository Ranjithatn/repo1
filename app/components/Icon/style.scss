


.fa-disabled {
  opacity: 0.6;
  cursor: not-allowed;
  color: #b5b5b5 !important;
}



