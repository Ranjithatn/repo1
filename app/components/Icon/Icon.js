import React from "react";
import "./style.scss";

const Icon = ({ icon, title, onClick, disabled=false, style={} }) => {
  return (
    <span
      className={ disabled ? "icon fa-disabled" : "icon" }
      title={title ? title : ""}
      style={style}
      onClick={(e) => typeof onClick === "function" && onClick(e)}
      disabled={ disabled }
    >
      <i className={"fa fa-" + icon} />
    </span>
  );
};

export default Icon;
