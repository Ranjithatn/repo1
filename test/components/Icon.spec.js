import React from "react";
import toJson from "enzyme-to-json";
import Icon from "../../app/components/Icon/Icon";
import Adapter from "enzyme-adapter-react-16";
import { configure, shallow, mount } from "enzyme";

configure({ adapter: new Adapter() });


describe( '<Icon />', () => {

	it("should render without crashing", () => {
		const component = mount(
			<Icon />
		);
		const tree = toJson(component);
		expect(tree).toMatchSnapshot();
	});


	it("has correct title prop", () => {
		const props = { title: 'User Icon' }
		const component = shallow(
			<Icon {...props} />
		);
		expect( component.props().title ).toBe('User Icon');
	});


	it("renders the component with all the given props", () => {
		const props = {
			icon: 'user',
			title: 'User Icon',
			onClick: () => {
				console.log("CLICKED!!!");
			}
		}
		const component = shallow(
			<Icon {...props} />
		);

		expect( component.find('span').props().title ).toBe(props.title);
		expect( component.find('i').props().className ).toBe('fa fa-user');
	});


	it("onClick class the onClick method", () => {
		const props = {
			icon: 'user',
			title: 'User Icon',
			onClick: () => {
				return true;
			}
		}
		const component = shallow(
			<Icon {...props} />
		);

		const click = component.find('span').props().onClick();

		expect( click ).toBe(true);

	});

});

