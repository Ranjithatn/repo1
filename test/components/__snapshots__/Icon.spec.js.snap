// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render without crashing 1`] = `
<Icon>
  <span
    className="icon"
    disabled={false}
    onClick={[Function]}
    style={Object {}}
    title=""
  >
    <i
      className="fa fa-undefined"
    />
  </span>
</Icon>
`;
